### PR TITLE
1.0.86

### PR DESCRIPTION
**Pull request by WeihanLi**
This pull request introduces major improvements to console color output handling in the codebase, particularly by enhancing the `ConsoleHelper` to use ANSI escape sequences for colored output when supported. It also adds comprehensive tests for these features, updates some build and workflow configurations, and improves documentation for extension methods.

**Console color output enhancements:**

* [`src/WeihanLi.Common/Helpers/ConsoleHelper.cs`](diffhunk://#diff-13ecca5852d1788521ae86ff2606df941b374f2b9759a06e430eab099eacec35R4-R12): Refactored all color output methods (`WriteWithColor`, `WriteLineWithColor`, `ErrorWriteWithColor`, `ErrorWriteLineWithColor`) to use ANSI escape sequences when supported, and added detection logic for ANSI color support across platforms. Also introduced helper methods for applying ANSI color codes and determining support. [[1]](diffhunk://#diff-13ecca5852d1788521ae86ff2606df941b374f2b9759a06e430eab099eacec35R4-R12) [[2]](diffhunk://#diff-13ecca5852d1788521ae86ff2606df941b374f2b9759a06e430eab099eacec35R70-R133) [[3]](diffhunk://#diff-13ecca5852d1788521ae86ff2606df941b374f2b9759a06e430eab099eacec35R266-R369)

**Testing improvements:**

* [`test/WeihanLi.Common.Test/HelpersTest/ConsoleHelperTest.cs`](diffhunk://#diff-ecde2703efec100e35adb2dab38769270f93737d3cb41f34705b68edb1ec3c20R1-R219): Added a new test suite for `ConsoleHelper`, covering all color output methods, edge cases (null/empty output), and the ANSI color support detection.

**Console output usage updates:**

* `samples/DotNetCoreSample/Program.cs`, `src/WeihanLi.Common/Helpers/BuildProcess.cs`: Updated usages of `Console.WriteLine` to use `ConsoleHelper.WriteLineWithColor` for improved readability and consistent color output in sample and build process logs. [[1]](diffhunk://#diff-a35159c348f331abf2243c5d714a35452462ceb5ae04a56466552cbcc68db357L9-R9) [[2]](diffhunk://#diff-99368b1709fc6c8e6bff3ceb45111634a141e34fe1c41ef00ee78623082521cdL241-R245) [[3]](diffhunk://#diff-99368b1709fc6c8e6bff3ceb45111634a141e34fe1c41ef00ee78623082521cdL298-R298) [[4]](diffhunk://#diff-99368b1709fc6c8e6bff3ceb45111634a141e34fe1c41ef00ee78623082521cdL327-R327)

**Build and workflow configuration:**

* `.github/workflows/default.yml`, `global.json`, `build/version.props`: Updated the build workflow to specify a build target, bumped the patch version, and upgraded the .NET SDK version to 10.0.100. [[1]](diffhunk://#diff-54411c2dc70cdd657fcf8a7bdebeff500fbb1245570abb83c22b3f62db03e71bL36-R36) [[2]](diffhunk://#diff-b239f49e3a9bbfba2c871517dae3af9071d49f69645ac44376fbd1812e4bb527L5-R5) [[3]](diffhunk://#diff-8df3cd354bc584349d04ad5675b33c042d8b99b741b8b95af394c55e0f5001bfL4-R4)

**Documentation improvements:**

* [`src/WeihanLi.Common/Extensions/CoreExtension.cs`](diffhunk://#diff-17c1585c0fdefbd09f6bbfc7fdc6414534049f8b6750334c02d2b53fb066da58L1063-R1065): Improved XML documentation for extension methods to clarify parameter purposes and return values. [[1]](diffhunk://#diff-17c1585c0fdefbd09f6bbfc7fdc6414534049f8b6750334c02d2b53fb066da58L1063-R1065) [[2]](diffhunk://#diff-17c1585c0fdefbd09f6bbfc7fdc6414534049f8b6750334c02d2b53fb066da58L1076-R1079)

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -42,12 +43,12 @@ steps:
 - script: dotnet build.cs
   displayName: 'Build Script'
   env:
-    Nuget__ApiKey: $(nugetApiKey)
+    NuGet__ApiKey: $(nugetApiKey)
```
[the remark this answers is not in the document]
Comment (WeihanLi) [reply azure-pipelines.yml:46]: intentional

File: test/WeihanLi.Common.Test/HelpersTest/ConsoleHelperTest.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    [Fact]
+    public void SupportsAnsiColorsReturnsBooleanValue()
+    {
+        // Act
+        var result = ConsoleHelper.SupportsAnsiColors();
+        
+        // Assert
+        Assert.IsType<bool>(result);
+    }
```
[the remark this answers is not in the document]
Comment (WeihanLi) [reply test/WeihanLi.Common.Test/HelpersTest/ConsoleHelperTest.cs]: can be removed

File: test/WeihanLi.Common.Test/HelpersTest/ConsoleHelperTest.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+}
```
[the remark this answers is not in the document]
Comment (WeihanLi) [reply test/WeihanLi.Common.Test/HelpersTest/ConsoleHelperTest.cs:38]: ANSI color is not friendly to test especially when running on CI terminal

File: src/WeihanLi.Common/Helpers/BuildProcess.cs [hunk 4]
```diff
@@ -333,15 +377,17 @@ private DotNetPackageBuildProcess(DotNetBuildProcessOptions options)
...
                     var nugetSource =  string.IsNullOrEmpty(_source) ? options.FallbackNuGetSourceFunc() : _source;
                     nugetSource = string.IsNullOrEmpty(nugetSource) ? string.Empty : $"--source {nugetSource}";
-                    var pushArgumenst = $" -k {_apiKey} --skip-duplicate {nugetSource}";
+                    var pushArguments = $" -k {_apiKey} --skip-duplicate {nugetSource}";
                     foreach (var file in Directory.GetFiles(options.ArtifactsPath, "*.nupkg"))
                     {
-                        var commandText = $"dotnet nuget push {file} {pushArgumenst}";
+                        var commandText = $"dotnet nuget push {file} {pushArguments}";
                         CommandExecutor.ExecuteCommandAndOutput(commandText).EnsureSuccessExitCode();
                     }
```
[the remark this answers is not in the document]
Comment (WeihanLi) [reply src/WeihanLi.Common/Helpers/BuildProcess.cs:419]: acceptable

File: src/WeihanLi.Common/Helpers/BuildProcess.cs [hunk 5]
```diff
@@ -257,10 +295,16 @@ private DotNetPackageBuildProcess(DotNetBuildProcessOptions options)
...
+                            string.Equals(EnvHelper.Val("GITHUB_ACTIONS"), "true", StringComparison.OrdinalIgnoreCase) &&
+                            !disableGitHubReport;
+                        var reportArguments = enableGitHubReport ? " --report-github" : string.Empty;
+
                         foreach (var project in options.TestProjects ?? [])
                         {
                             CommandExecutor.ExecuteCommandAndOutput(
-                                $"dotnet test --collect:\"XPlat Code Coverage;Format=cobertura,opencover;ExcludeByAttribute=ExcludeFromCodeCoverage,Obsolete,GeneratedCode,CompilerGeneratedAttribute\" {project}"
+                                $"dotnet test --project {project}{reportArguments}"
                                 ).EnsureSuccessExitCode();
```
[the remark this answers is not in the document]
Comment (WeihanLi) [reply src/WeihanLi.Common/Helpers/BuildProcess.cs]: intentional, not sure what's the best practice for using microsoft testing platform to collect test coverage, keep it as for now, will update in later versions
